### PR TITLE
feat: definitions

### PR DESCRIPTION
**Pull request by vankop**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

#49 

### Breaking Changes

No 

### Additional Info

Nothing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +35,7 @@
     "dist"
   ],
   "dependencies": {
+    "@types/json-schema": "^7.0.3",
```
Comment (evilebottnawi) [inline package.json]: Not sure it is good idea. Some projects doesn't use ts so they don't need extra libraries in deps

**Comment by vankop** [comment]
Снимок экрана 2019-09-02 в 16 02 04

It is small and we can fix version

**Comment by evilebottnawi** [comment]
@vankop we need avoid extra deps, because this is used inside webpack

**Comment by vankop** [comment]
> @vankop we need avoid extra deps, because this is used inside webpack

so... copy paste?

**Comment by evilebottnawi** [comment]
@vankop yep, copy/paste

**Comment by vankop** [comment]
I was curious to create babel plugin for extracting JSON schema definitions on build, but looks like it will be overengineering for `schema-utils` purposes

**Comment by vankop** [comment]
@evilebottnawi could you review this, please?

**Comment by evilebottnawi** [comment]
@vankop near future, some busy

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+    postFormatter: PostFormatter
+  }
+}
+
+declare var validate: {
+  (schema: Schema, options: Array<object> | object, configuration?: Partial<SchemaUtils.ValidateErrorConfiguration>): void;
+  ValidateError: typeof SchemaUtils.ValidateError
+}
+
+export = validate;
```
Comment (evilebottnawi) [inline index.d.ts]: Why don't add some types in devDeps?

Comment (vankop) [reply index.d.ts]: which one?

Comment (evilebottnawi) [reply index.d.ts]: `@types/json-schema`

Comment (vankop) [reply index.d.ts]: Hmmm, I just followed your previous suggestions -( Ok i will revert this and move this to dev dependency

Comment (evilebottnawi) [reply index.d.ts]: @vankop Sometimes I have a lot issues and lose context, sorry for that, all make mistakes :smile:

Comment (vankop) [reply index.d.ts]: It is fine

**Comment by vankop** [comment]
I think not since it design to create definitions from code and not vice versa.

**Comment by evilebottnawi** [comment]
@vankop hm, what about this https://github.com/sindresorhus/execa/blob/master/index.test-d.ts?

**Comment by vankop** [comment]
I will take a look, if you think we need some tests

**Comment by evilebottnawi** [comment]
@vankop let's open new issue about this, i think here is all good

**Pull request merged**